src/phpDocumentor/Descriptor/Class_.php

```diff
@@ -0,0 +1,223 @@
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor\Descriptor;
+use phpDocumentor\Reflection\DocBlock;
+use phpDocumentor\Reflection\Element;
+use phpDocumentor\Reflection\Fqsen;
+
+/**
+ * Descriptor representing a Class.
+ */
+final class Class_ implements Element
+{
+    /**
+     * @var Fqsen Full Qualified Structural Element Name
+     */
+    private $fqsen;
+
+    /**
+     * @var DocBlock | Null
+     */
+    private $docBlock = null;
+
+    /** @var boolean $abstract Whether this is an abstract class. */
+    private $abstract = false;
+
+    /** @var boolean $final Whether this class is marked as final and can't be subclassed. */
+    private $final = false;
+
+    /**
+     * @var Class_ $parent The class this class is extending.
+     */
+    private $parent = null;
+
+    /** @var Interface_[] $implements References to interfaces that are implemented by this class. */
+    private $implements = array();
+
+    /** @var Constant[] $constants References to constants defined in this class. */
+    private $constants = array();
+
+    /** @var Property[] $properties References to properties defined in this class. */
+    private $properties = array();
+
+    /** @var Method[] $methods References to methods defined in this class. */
+    private $methods = array();
+
+    /** @var Trait_[] $usedTraits References to traits consumed by this class */
+    private $usedTraits = array();
+
+    /**
+     * Initializes the all properties representing a collection with a new Collection object.
+     *
+     * @param Fqsen $fqsen
+     * @param DocBlock $docBlock
+     * @param bool $abstract
+     * @param bool $final
+     */
+    public function __construct(Fqsen $fqsen, DocBlock $docBlock = null, Class_ $parent = null, $abstract = false, $final = false)
+    {
+        $this->fqsen = $fqsen;
+        $this->parent = $parent;
+        $this->docBlock = $docBlock;
+        $this->abstract = $abstract;
+        $this->final = $final;
+    }
+
+    /**
+     * Returns true when this method is final. Otherwise returns false.
+     *
+     * @return bool
+     */
+    public function isFinal()
+    {
+        return $this->final;
+    }
+
+    /**
+     * Returns true when this method is abstract. Otherwise returns false.
+     *
+     * @return bool
+     */
+    public function isAbstract()
+    {
+        return $this->abstract;
+    }
+
+    /**
+     * Returns the Class_ this class is extending if available.
+     *
+     * @return NUll|Class_
+     */
+    public function getParent()
+    {
+        return $this->parent;
+    }
+
+    /**
+     * Returns the interfaces this class is implementing.
+     *
+     * @return Interface_[]
+     */
+    public function getInterfaces()
+    {
+        return $this->implements;
+    }
+
+    /**
+     * Add interface this class is implementing.
+     *
+     * @param Interface_ $interface
+     */
+    public function addInterface(Interface_ $interface)
```

Comment:
This method should accept a FQSEN object and not an Interface_ object since it represents a link to another object; the View component in phpDocumentor will make the necessary connections. This is done because it represents the structure of the project, it will make caching more efficient and will reduce errors when updating an existing structure. Should a class be deleted from the project we will not lose references.